applications/IgaApplication/custom_modelers/nurbs_geometry_modeler_sbm.h

```diff
@@ -0,0 +1,129 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Nicolo' Antonelli
+//                   Andrea Gorgi
+//
+
+#if !defined(KRATOS_NURBS_GEOMETRY_MODELER_SBM_H_INCLUDED )
+#define  KRATOS_NURBS_GEOMETRY_MODELER_SBM_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/model_part.h"
+#include "nurbs_geometry_modeler.h"
+#include "geometries/nurbs_volume_geometry.h"
+#include "geometries/nurbs_surface_geometry.h"
+#include "geometries/nurbs_shape_function_utilities/nurbs_surface_refinement_utilities.h"
+#include "geometries/brep_curve_on_surface.h"
+#include "utilities/nurbs_utilities/snake_sbm_utilities.h"
+
+namespace Kratos {
+
+class KRATOS_API(IGA_APPLICATION) NurbsGeometryModelerSbm
+    : public NurbsGeometryModeler
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    KRATOS_CLASS_POINTER_DEFINITION( NurbsGeometryModelerSbm );
+
+    typedef std::size_t IndexType;
+    typedef std::size_t SizeType;
+    typedef Node NodeType;
+
+    typedef Geometry<NodeType> GeometryType;
+    typedef typename GeometryType::Pointer GeometryPointerType;
+
+    typedef NurbsSurfaceGeometry<3, PointerVector<NodeType>> NurbsSurfaceGeometryType;
+    typedef typename NurbsSurfaceGeometryType::Pointer NurbsSurfaceGeometryPointerType;
+
+    typedef NurbsVolumeGeometry<PointerVector<NodeType>> NurbsVolumeGeometryType;
+    typedef typename NurbsVolumeGeometryType::Pointer NurbsVolumeGeometryPointerType;
+
+    typedef PointerVector<Node> ContainerNodeType;
+    typedef PointerVector<Point> ContainerEmbeddedNodeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    NurbsGeometryModelerSbm()
```

Comment:
those defaults maybe they can be removed

Reply:
If I'm not wrong, the compiler will add it if not in there.